From ce905a154d8ad013b020b73d81c4275ab61679e3 Mon Sep 17 00:00:00 2001
From: Jai Luthra <j-luthra@ti.com>
Date: Thu, 8 Dec 2022 17:23:50 +0530
Subject: [PATCH 5/5] media: ov5640: Handle delays when no reset_gpio set

Some RPi-compatible module manufacturers [1] don't expose the RESETB
gpio of the sensor directly through the connector. Instead wiring ~PWDN
as a proxy reset with appropriate delays.

In such cases, reset_gpio will not be available to the driver, but it
will still be toggled when the sensor is powered on, and thus we should
still honor the wait time of >= 5ms + 1ms + 20ms [2] before
attempting any i/o operations over SCCB.

[1] https://digilent.com/reference/_media/reference/add-ons/pcam-5c/pcam_5c_sch.pdf
[2] https://cdn.sparkfun.com/datasheets/Sensors/LightImaging/OV5640_datasheet.pdf

Signed-off-by: Jai Luthra <j-luthra@ti.com>
---
 drivers/media/i2c/ov5640.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/i2c/ov5640.c b/drivers/media/i2c/ov5640.c
index 6300e4a445f6..84da0f8546a5 100644
--- a/drivers/media/i2c/ov5640.c
+++ b/drivers/media/i2c/ov5640.c
@@ -2414,6 +2414,7 @@ static int ov5640_set_power_on(struct ov5640_dev *sensor)
 
 	ov5640_reset(sensor);
 	ov5640_power(sensor, true);
+	usleep_range(26000, 30000);
 
 	ret = ov5640_init_slave_id(sensor);
 	if (ret)
-- 
2.30.2

